File: auto-provision/task/scripts/sendmail.py
#
# Apache v2 license
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
#
import smtplib
from email.header import Header
from email.mime.text import MIMEText
from email.mime.application import MIMEApplication
from email.mime.multipart import MIMEMultipart
from email.utils import parseaddr, formataddr

import pathlib
import click

#mail_host = "smtp.intel.com"
#mail_port = 25


@click.command()
@click.option('-s', default='xxxx@intel.com', help='sender')
@click.option('-r', default='xxxx@intel.com', help='receiver')
@click.option('-c', default='this is auto email form intel', help='message content')
@click.option('-m', default='', help='smtp server')
@click.option('-p', default='25', help='smtp port')
@click.option('-f', default=None, help='attachment path')
def send_email(s, r, c, m ,p, f):
    
    r = r.replace(" ", "")
    r = r.split(",")
    c = c.replace("===", " ")
    title = '[Notice]Workloads auto deployment test tools execution report'
    if f:
        message = MIMEMultipart()
        part_text = MIMEText(c)
        message.attach(part_text)
        attach = MIMEApplication(open(f, 'rb').read())
        attach.add_header('Content-Disposition', 'attachment',
                          filename=pathlib.Path(f).name)
        message.attach(attach)
    else:
        message = MIMEText(c, 'html', 'utf-8')
    #message['From'] = "{}".format(s)
    message["From"] = Header(s, 'utf-8')
    #message["From"] = __format_addr('NPG-WiE-Benchmarking-Report@intel.com')
    message['To'] = ",".join(r)
    message['Subject'] = title
    try:
        smtpObj = smtplib.SMTP()
        smtpObj.connect(m, int(p))
        #smtpObj.sendmail(sender, receivers, message.as_string())
        smtpObj.sendmail(s, r, message.as_string())
        print("mail has been send successfully.")
    except smtplib.SMTPException as e:
        print(e)


send_email()

File: auto-provision/task/test/sendmail.go
/*
Apache v2 license
Copyright (C) 2023 Intel Corporation
SPDX-License-Identifier: Apache-2.0
*/
package model

import (
	"fmt"
	"net/smtp"
	"strings"
)

func SendToMail(userName, password, mailServer, to, subject, body, mailType string) error {

	// 拼接消息体
	var contentType string
	if mailType == "html" {
		contentType = "Content-Type: text/" + mailType + "; charset=UTF-8"
	} else {
		contentType = "Content-Type: text/plain" + "; charset=UTF-8"
	}
	msg := []byte("To: " + to + "\r\nFrom: " + userName + "\r\nSubject: " + subject + "\r\n" + contentType + "\r\n\r\n" + body)

	// msg 内容输出查看
	fmt.Println("To: " + to + "\r\n" +
		"From: " + userName + "\r\n" +
		"Subject: " + subject + "\r\n" +
		"" + contentType + "\r\n\r\n" +
		"" + body)

	// 进行身份认证
	hp := strings.Split(mailServer, ":")
	auth := smtp.PlainAuth("", userName, password, hp[0])

	sendTo := strings.Split(to, ";")
	err := smtp.SendMail(mailServer, auth, userName, sendTo, msg)
	return err
}

func main() {
	var userName = "***@163.com"
	var password = "***"
	var mailServer = "smtp.163.com:25"
	var to = "**@qq.com"
	var subject = "犀点意象安全团队"

	body := `
		您的服务存在异常，请查收！！！
		`
	bodyHtml := `<html>
					<body>
						<h1>您的服务存在异常</h1>
					</body>
				</html>
				`
	fmt.Println("send email")
	var err error
	err = SendToMail(userName, password, mailServer, to, subject, body, "")
	if err != nil {
		fmt.Println("Send text-mail error!")
		fmt.Println(err)
	} else {
		fmt.Println("Send text-mail success!")
	}
	err = SendToMail(userName, password, mailServer, to, subject, bodyHtml, "html")
	if err != nil {
		fmt.Println("Send html-mail error!")
		fmt.Println(err)
	} else {
		fmt.Println("Send html-mail success!")
	}
}
